hashtable/open_addressing/hashmap.go

```diff
@@ -0,0 +1,161 @@
+package open_addressing
+
+const (
+	// KeyNotFound is a constant indicates that
+	// a key cannot be found in the HashMap.
+	KeyNotFound      = 0
+	defaultCapacity  = 8
+	defaultPrime     = 7
+	upperLoadFactor  = 0.75
+	lowerLoadFactor  = 0.2
+	growShrinkFactor = 2
+)
+
+type node struct {
+	key     int
+	value   int
+	deleted bool
+}
+
+// HashMap is a struct for storing key-value data.
+type HashMap struct {
+	items []*node
+	size  int
+	prime int
```

Comment:
I don't think we need to save this property.

Reply:
I use a prime number in my 2nd hash function, and because it's dependent on the bucket array capacity, I store it so every call to my hash function will just take O(1). But as you pointed above, it takes O(n^2) when rehashing the Hash Table.